src/v/kafka/server/datalake_throttle_manager.cc

```diff
@@ -39,28 +39,30 @@ void datalake_throttle_manager::merge_producer_maps(
     }
 }
 
-datalake_throttle_manager::backlog_status
-datalake_throttle_manager::backlog_status::operator+(
-  const datalake_throttle_manager::backlog_status& o) const {
-    return datalake_throttle_manager::backlog_status{
-      .partitions_backlog_limit_breached
-      = partitions_backlog_limit_breached + o.partitions_backlog_limit_breached,
+datalake_throttle_manager::status datalake_throttle_manager::status::operator+(
+  const datalake_throttle_manager::status& o) const {
+    return datalake_throttle_manager::status{
+      .max_shares_assigned = max_shares_assigned || o.max_shares_assigned,
+      .overdue_translation_partition_count
+      = overdue_translation_partition_count
+        + o.overdue_translation_partition_count,
       .partitions_translation_blocked = partitions_translation_blocked
                                         + o.partitions_translation_blocked};
 }
 
-bool datalake_throttle_manager::backlog_status::has_no_issues() const {
-    return partitions_backlog_limit_breached == 0
-           && partitions_translation_blocked == 0;
+bool datalake_throttle_manager::status::has_no_issues() const {
```

Comment:
Nit: "has no issues" is kind of misleading now. Maybe invert and rename to `needs_throttling`?